### PR TITLE
credentials-controller: fetch credentials using cloud-credentials-operator

### PR DESCRIPTION
**Pull request by snir911**
flow is:

credentials-controller creates credentialsRequests for all supported providers upon
KataConfig+PeerPods creation ->
CCO creates cco-secret in response to provider's credentialsRequests ->
credentials-controller triggers by cco-secret creation and translates cco-secrets
to peer-pods-secret owned by the cco-secret

at deletion of KataConfig:
credentials-controller delets credentialsRequests ->
CCO deletes the cco-secret ->
GC deletes owned peer-pods-secret

to test:
```
quay.io/snir/osc-operator-catalog:v9.9.114
quay.io/snir/osc-operator-bundle:v9.9.114
quay.io/snir/osc-operator:v9.9.114

```

**avoiding OOM patch introduced by #375 , remove it**

### CODE REVIEW

File: controllers/credentials_controller.go [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+		r.Log.Info("secretMapping: peerPodsSecret data is uninitialized")
+		return
+	}
+
+	if len(ccoSecret.Data) == 0 {
+		r.Log.Info("secretMapping: ccoSecret data is uninitialized or empty")
+		return
+	}
+
+	for ccoKey, ppKey := range ccoSecret.Data {
```
Comment (bpradipt) [inline controllers/credentials_controller.go:158]: have you tried this strings.ToUpper()?

File: controllers/credentials_controller.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+		if ccoToPp[ccoKey] != "" {
+			peerPodsSecret.Data[ccoToPp[ccoKey]] = ppKey
+		}
+	}
+}
+
+func (r *SecretReconciler) newOwnedPeerPodsSecret(ownedSecret *corev1.Secret) *corev1.Secret {
```
Comment (bpradipt) [inline controllers/credentials_controller.go:165]: Can you pls add some description for this method and how the lifecycle (key rotation etc) will get handled? A reference to doc will also be good

File: controllers/credentials_controller.go (hunk 1 above)
Comment (snir911) [reply controllers/credentials_controller.go:158]: yes, it seems like it's not 100% mapped this way e.g. azure_resourcegroup, not sure about other providers fields

Comment (bpradipt) [reply controllers/credentials_controller.go:158]: I see :-( . can you add a comment before the loop explaining this. Will be useful in future when we relook at this code

File: controllers/credentials_controller.go [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+)
+
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=secrets/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core,resources=secrets/finalizers,verbs=update
+//+kubebuilder:rbac:groups=cloudcredential.openshift.io,resources=credentialsrequests,verbs=create;delete
+
+// TODO: reduce secret's RBAC if possible
+
+// Reconcoiles cco-secret/peer-pods secrets based on the secretsFilterPredicate and maps the cco-secret
```
Comment (bpradipt) [inline controllers/credentials_controller.go:64]:
```suggestion
// Reconciles cco-secret/peer-pods secrets based on the secretsFilterPredicate and maps the cco-secret
```

File: controllers/credentials_controller.go [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(secretsFilterPredicate())).
+		Watches(&kataconfigurationv1.KataConfig{},
+			&KataConfigHandler{r}).
+		Complete(r)
+}
+
+func secretsFilterPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetNamespace() == "openshift-sandboxed-containers-operator" && e.ObjectNew.GetName() == credentialsRequestSecretRefName
```
Comment (bpradipt) [inline controllers/credentials_controller.go:123]: nit: Better to define the namespace name as a constant

**Comment by bpradipt** [comment]
Nice work @snir911 
Since this doesn't rely on MCO, wondering if it's possible to add unit tests for this workflow leveraging envtest?

**Review by littlejawa — COMMENTED** [review]
lgtm from the code standpoint - didn't run a test as I don't have a the environment to test with.
But I see it is still marked as "do not merge", is it ready to review?

**Comment by snir911** [comment]
Thanks @littlejawa, yes it's ready, there's a problem to have the pod actually running regardless to this PR ATM, however, it does fetch and create the secret for you, i'll remove the DNM

**Comment by vvoronko** [comment]
pre-merge tested e2e on AWS

**Comment by vvoronko** [comment]
pre-merge tested on Azure, generated secret has 2 aprameters that should be moved into configmap:
AZURE_REGION and AZURE_RESOURCE_GROUP

**Comment by snir911** [comment]
updated, according to Victor's comment it's now skip setting AZURE_REGION and AZURE_RESOURCE_GROUP

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @snir911

**Pull request merged**